src/toil/test/provisioners/cgcloud/provisionerTest.py

```diff
@@ -265,3 +266,44 @@ def testAutoScaledCluster(self):
     @integrative
     def testAutoScaledSpotCluster(self):
         self._test(autoScaled=True, spotInstances=True)
+
+
+restartScript = """\"from toil.job import Job
+import argparse
+import os
+
+
+def f0(job):
+    if 'FAIL' in os.environ:
+        raise RuntimeError('failed on purpose')
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    Job.Runner.addToilOptions(parser)
+    options = parser.parse_args()
+    i = Job.Runner.startToil(Job.wrapJobFn(f0), options)
+\""""
+
+
+class CGCloudRestartTest(AbstractCGCloudProvisionerTest):
+
+    def _getScript(self):
+        self.scriptName= "restartScript.py"
+        self._leader('echo %s > %s' % (restartScript, 'restartScript.py'), shell=True)
```

Comment:
If not lets debug that together. The `echo` technique can break, if, for example, you had double quotes in your script.